src/main/webapp/common.js

```diff
@@ -109,9 +110,81 @@ async function displayCurrentAddress() {
 }
 
 /**
- * Method that builds for every page with a "menu-container div" a navigable menu.
+ * If checkboxId element is checked, the function will update the address box addressId to user's
+ * current address so that user can check its location accuracy.
+ */
+function updateAddressFromGeolocation(checkboxId, addressId) {
+  addressBox = document.getElementById(addressId);
+  if (document.getElementById(checkboxId).checked) {
+    // update the address box to show the address found using Geolocation
+    if (navigator.geolocation) {
+        // if location was retrieved successfully, call setLatLng, else alert in geolocationError
+        getLocation().then((position) => {
+          setLatLng(position);
+          getAddressFromLatLng(addressId)
+        }).catch((error) => geolocationError(error));
+    } else {
+      // clear the address box if location is not supported 
+      addressBox.value = ""; 
+      alert("Geolocation is not supported by this browser!");
+    }
+  }
+}
+
+function getLocation() {
+  // getCurrentPosition immediately returns when called; then asynchronously attempts to
+  // obtain the current location of the device;
+  return new Promise(function(resolve, reject) {
+          navigator.geolocation.getCurrentPosition(resolve, reject);
+  });
+}
+
+/**
+ * Alerts user about the error message.
+ */
+function geolocationError(error) {
+  switch(error.code) {
+    case error.PERMISSION_DENIED:
+      alert("Please allow access to your location!");
+      break;
+    case error.POSITION_UNAVAILABLE:
+      alert("Location information is unavailable!");
+      break;
+    case error.UNKNOWN_ERROR:
+      alert("Error occured!");
+      break;
+  }
+}
+
+/**
+ * Sets latitude and longitude elements to user's coordinates using Geolocation API 
  */
+function setLatLng(position) {
+  document.getElementById('latitude').value = position.coords.latitude;
+  document.getElementById('longitude').value = position.coords.longitude;
+}
 
+/**
+ * Returns user's address using reverse geocoding on latitude and longitude elements
+ */
+function getAddressFromLatLng(addressId) {
+  addressBox = document.getElementById(addressId);
+  geocodingUrl = `https://maps.googleapis.com/maps/api/geocode/json?latlng=${document.getElementById("latitude").value},${document.getElementById("longitude").value}&key=${GoogleMapsApiKey}`;
+  fetch(geocodingUrl).then(response => response.json()).then((result) => {
+    
+    if (result.status != "OK") {
+      // there was an error when getting the address from the coordinates
+      alert("Location information is unavailable!");
+    } else {
+      // return the best address found 
+      addressBox.value = result.results[0].formatted_address;
+    }
+  });
+}
```

Comment:
Can you split this function as well - first one gets the values from the DOM, and the second one handles the API call and returning the results as a string.

Then you can unit-test :)